src/App.tsx

```diff
@@ -3,35 +3,119 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api/api';
 
 interface State {
+  todos: Todo[],
   selectedUserId: number;
+  input: string,
+  select: string,
 }
 
-class App extends React.Component<{}, State> {
+export class App extends React.Component<{}, State> {
   state: State = {
+    todos: [],
     selectedUserId: 0,
+    input: '',
+    select: '',
+  };
+
+  async componentDidMount() {
+    const todosFromServer = await getTodos();
+
+    this.setState({
+      todos: todosFromServer,
+    });
+  }
+
+  handleClear = () => {
+    this.selectUser(0);
+  };
+
+  selectUser = (userId: number) => {
+    this.setState({
+      selectedUserId: userId,
+    });
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      input: value,
+    });
+  };
+
+  handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      select: value,
+    });
+  };
+
+  preparedTododos = () => {
+    const { todos, input, select } = this.state;
+
+    const filteredTodos = todos
+      .filter(todo => (
+        todo.title.toLowerCase()
+          .includes(input.toLowerCase())
+      ));
```

Comment:
what if input is empty? so no need to do this filtering;
also no need to do `input.toLowerCase()` on every iteration, you can do it before filtering